operations/rollout-operator/rollout-operator.libsonnet

```diff
@@ -0,0 +1,380 @@
+  local serviceAccount = $.core.v1.serviceAccount,
+  local servicePort = $.core.v1.servicePort,
+  local validatingWebhook = $.admissionregistration.v1.validatingWebhook,
+  local validatingWebhookConfiguration = $.admissionregistration.v1.validatingWebhookConfiguration,
+
+  _config+:: {
+    rollout_operator_enabled: false,
+
+    // Configure the rollout operator to accept webhook requests made as part of scaling
+    // statefulsets up or down. This allows the rollout operator to ensure that stateful
+    // components (ingesters, store-gateways) are scaled up or down safely.
+    enable_rollout_operator_webhook: false,
+
+    // Specifically enable zone aware pod distruption budget support, including the rollout-operator webhooks.
+    // Note that zpdb support is enabled by default if webhooks are enabled.
+    // See disable_rollout_operator_eviction_webhook for additional configuration options.
+    enable_zpdb: false,
+
+    // Disable the deployment of the pod eviction webhook.
+    // This will only have an effect if webhooks are enabled.
+    // Note that the eviction webhook is safe to be deployed even if there are no zone pdb configurations.
+    // Pod evictions not handled by a zpdb will be allowed and contine to be handled by any other pdb configurations.
+    disable_rollout_operator_eviction_webhook: false,
+
+    // Include the custom resource definiton for the zpdb.
+    zpdb_custom_resource_definition_enabled: true,
+
+    // Configure the rollout operator to enable support for ReplicaTemplates
+    rollout_operator_replica_template_access_enabled: false,
+
+    // Include the custom resource definiton for ReplicaTemplates.
+    replica_template_custom_resource_definition_enabled: false,
+
+    // Ignore_rollout_operator_*_webhook_failures will set the rollout-operator to ignore
+    // webhook failures. Useful during a rollout to a new cell, where rollout-operator service
+    // is still not created, as the webhook might be created before the service, and that could
+    // block other operations that would block the service creation.
+    ignore_rollout_operator_no_downscale_webhook_failures: false,
+    ignore_rollout_operator_prepare_downscale_webhook_failures: false,
+    ignore_rollout_operator_zpdb_eviction_webhook_failures: false,
+    ignore_rollout_operator_zpdb_validation_webhook_failures: false,
+
+    // Ignore these labels used for controlling webhook behavior when creating services.
+    service_ignored_labels+:: ['grafana.com/no-downscale', 'grafana.com/prepare-downscale'],
+  },
+
+  // implicitly enable webhooks if the zpdb or replica template support has been requested
+  local enableWebhooks =
+    $._config.enable_zpdb ||
+    $._config.rollout_operator_replica_template_access_enabled ||
+    $._config.enable_rollout_operator_webhook,
+
+  // implicitly enable rollout-operator if the webhooks have been requested
+  local enableRolloutOperator = enableWebhooks || $._config.rollout_operator_enabled,
```

Comment:
Similar to above - if webhooks are enabled but the rollout-operator as a whole isn't, then I'd either fail with an assertion message or ignore the webhooks and create nothing.